lib/vrl/stdlib/src/parse_json.rs

```diff
@@ -7,6 +10,63 @@ fn parse_json(value: Value) -> Resolved {
     Ok(value)
 }
 
+fn parse_json_with_depth(value: Value, max_depth: Value) -> Resolved {
+    let bytes = value.try_bytes()?;
+    let parsed_depth = positive_int_depth(max_depth)?;
+
+    let raw_value: Box<RawValue> =
+        serde_json::from_slice::<_>(&bytes).map_err(|e| format!("unable to read json: {}", e))?;
+
+    let res = parse_once_with_depth(raw_value, parsed_depth)
+        .map_err(|e| format!("unable to parse json with max depth: {}", e))?;
+    Ok(Value::from(res))
+}
+
+fn positive_int_depth(value: Value) -> std::result::Result<i64, ExpressionError> {
+    let res = value.try_integer()?;
+    if !(1..=128).contains(&res) {
+        Err(ExpressionError::from(format!(
+            "max_depth value should be greater than 0 and less than 128, got {}",
```

Comment:
Could we mention this range restriction in the docs for this config option? I'm also wondering why we need to cap it at all?

Reply:
The lower cap is `1` because it is pointless to use anything lover, `data = parse_json!(.message, max_depth: 0)` equals to `data = .message`.

The upper cap is 128 because serde_json [has the same](https://github.com/serde-rs/json/blob/4d57ebeea8d791b8a51c229552d2d480415d00e6/json/src/de.rs#L111) recursion limit by default.

I will add the info to the doc.